### ISSUE TITLE
Azure - Tests - Provision.sh display help with no options

### ISSUE THREAD

**Issue by jomalsan** [cloud-custodian/cloud-custodian#4705]
For `tools/c7n_azure/tests/templates/provision.sh` I expected that the default behavior with no arguments would be displaying a help menu. Turns out it deploys all of the arm templates to your current subscription.

Feature request to add a help menu as a default behavior and then include a flag such as `--all` to deploy all of the templates

**Comment by stefangordon** [comment]
Yeah the default usage is just `provision.sh` and `cleanup.sh` but I could see an `all` option being good.  Would break anyone depending on it, but I suspect that is mostly just our functional test pipeline.
